ArduCopter/AP_Arming.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <AP_Arming/AP_Arming.h>
+
+class AP_Arming_Copter : public AP_Arming
+{
+public:
+    AP_Arming_Copter(const AP_AHRS_NavEKF &ahrs_ref, const AP_Baro &baro, Compass &compass,
+                     const AP_BattMonitor &battery, const AP_InertialNav_NavEKF &inav,
+                     const AP_InertialSensor &ins) :
+        AP_Arming(ahrs_ref, baro, compass, battery),
+        inertial_nav(inav),
+        _ins(ins),
+        ahrs_navekf(ahrs_ref)
+        {
+    }
+
+    void update(void);
+    bool all_checks_passing(bool arming_from_gcs);
+    void pre_arm_rc_checks();
+
+protected:
+
+    bool pre_arm_checks(bool display_failure) override;
+    bool pre_arm_gps_checks(bool display_failure);
+    bool pre_arm_ekf_attitude_check();
+    bool pre_arm_terrain_check(bool display_failure);
+    bool pre_arm_proximity_check(bool display_failure);
+    bool arm_checks(bool display_failure, bool arming_from_gcs);
+
+    // NOTE! the following check functions *DO* call into AP_Arming:
+    bool ins_checks(bool display_failure) override;
+    bool compass_checks(bool display_failure) override;
+
+    // NOTE! the following check functions *DO NOT* call into AP_Arming!
+    bool gps_checks(bool display_failure);
+    bool fence_checks(bool display_failure);
+    bool board_voltage_checks(bool display_failure);
+    bool parameter_checks(bool display_failure);
+    bool pilot_throttle_checks(bool display_failure);
+    bool barometer_checks(bool display_failure);
+    bool rc_calibration_checks(bool display_failure);
+
+    void set_pre_arm_check(bool b);
+    void set_pre_arm_rc_check(bool b);
+
+    enum HomeState home_status() const override;
+
+private:
+
+    void gcs_send_text(MAV_SEVERITY severity, const char *str);
+
+    const AP_InertialNav_NavEKF &inertial_nav;
+    const AP_InertialSensor &_ins;
```

Comment:
My patch.  Severe stupidity can't be ruled out, considering where we use it we rename it to `ins`....